### PR TITLE
Added node.Retired

### LINKED ISSUE #3135: Adding support for node.retired and node.retired_reason
This is a feature that I would like to have access too.

Gophercloud code being changed: https://github.com/gophercloud/gophercloud/blob/d8427580463e54c7738e5499ce234417ce8223ff/openstack/baremetal/v1/nodes/results.go#L94

The fields as defined in Ironic: https://github.com/openstack/ironic/blob/1e52143f07266b53ad47d49a5be04cbf85cccc8b/ironic/objects/node.py#L178

### PR DESCRIPTION
**Pull request by Sharpz7**
Prior to starting a PR, please make sure you have read our
[contributor tutorial](https://github.com/gophercloud/gophercloud/tree/master/docs/contributor-tutorial).

Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

Fixes #3135 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/ironic/blob/1e52143f07266b53ad47d49a5be04cbf85cccc8b/ironic/objects/node.py#L178

### CODE REVIEW

File: openstack/baremetal/v1/nodes/results.go [hunk 1]
```diff
@@ -261,6 +261,13 @@ type Node struct {
 
 	// The UTC date and time when the last inspection was finished, ISO 8601 format. May be “null” if inspection hasn't been finished yet.
 	InspectionFinishedAt *time.Time `json:"inspection_finished_at"`
+
+	// Whether the node is retired. A Node tagged as retired will prevent any further
+	// scheduling of instances, but will still allow for other operations, such as cleaning, to happen
+	Retired bool `json:"retired"`
+
+	// Reason the node is marked as retired.
+	RetiredReason string `json:"retired_reason"`
```
Comment (EmilienM) [inline openstack/baremetal/v1/nodes/results.go:270]: please add tests into fixtures: `https://github.com/gophercloud/gophercloud/blob/f6262230510eafd738530d900a389da39fa7a977/openstack/baremetal/v1/nodes/testing/fixtures_test.go`

Comment (Sharpz7) [reply openstack/baremetal/v1/nodes/results.go:270]: Can do :))

**Comment by EmilienM** [comment]
@Sharpz7 unfortunately, the v1 branch doesn't take semver:minor kind of API changes, because it's in maintenance/bugfix only since very recently. We can make exceptions though, but we really encourage folks to migrate to v2 (we have a [manual](https://github.com/gophercloud/gophercloud/blob/master/docs/MIGRATING.md) for that).

**Review by EmilienM — APPROVED** [review]

**Comment by jayofdoom** [comment]
Thanks for the exception @EmilienM -- we'll connect with the openstack-prometheus-exporter community to see if we can help with that.

**Comment by Sharpz7** [comment]
Thank you @EmilienM for your help :))

**Comment by jayofdoom** [comment]
FWIW; there seems to be some evidence that field adding has been allowed for v1 backporting in the past: https://github.com/gophercloud/gophercloud/commit/7d47899eb9ceefcdf1f428626eb7458d4d60d8a4

**Pull request merged**